shared/network/src/p2p_model_sharing.rs

```diff
@@ -91,123 +84,107 @@ impl PeerManagerHandle {
     /// Peers that are available to request the model to
     available_peers: VecDeque<NodeId>,
     /// A map for the peer's blob ticket to their errors
-    /// A node could success sending the blob ticket but fail in the middle of the download so we differentiate between the two
-    /// Node -> (blob_ticket_request_errors, blob_ticket_downloads_errors)
-    errors_per_peers: HashMap<NodeId, (u8, u8)>,
+    errors_per_peers: HashMap<NodeId, u8>,
     /// Max errors we tolerate for a peer to share a parameter blob ticket
     max_errors_per_peer: u8,
-    /// Max errors we tolerate for a peer to fail in the middle of the download of a parameter blob ticket
-    max_retries_per_peer: u8,
 }
 
 impl PeerManagerActor {
-    pub fn new(max_errors_per_peer: u8, max_retries_per_peer: u8) -> Self {
+    pub fn new(max_errors_per_peer: u8) -> Self {
         Self {
             available_peers: VecDeque::new(),
             errors_per_peers: HashMap::new(),
             max_errors_per_peer,
-            max_retries_per_peer,
         }
     }
 
     fn handle_message(&mut self, message: PeerCommand, cancellation_token: CancellationToken) {
         match message {
             PeerCommand::SetPeers { peers } => {
                 self.available_peers = VecDeque::from(peers);
-                let errors_per_peers_vec =
-                    self.available_peers.iter().map(|peer| (*peer, (0u8, 0u8)));
+                let errors_per_peers_vec = self.available_peers.iter().map(|peer| (*peer, 0_u8));
                 self.errors_per_peers = HashMap::from_iter(errors_per_peers_vec);
 
-                debug!(
+                info!(
                     "Updated peer list: {} peers available to ask for the model parameters",
                     self.available_peers.len()
                 );
             }
             PeerCommand::GetPeer { reply } => {
+                info!("AVAILABLE PEERS: {:?}", self.available_peers);
                 let peer = if let Some(peer) = self.available_peers.pop_front() {
-                    debug!("Selected peer {peer} to ask for the model parameters");
+                    info!("Selected peer {peer} to ask for the model parameters");
                     Some(peer)
                 } else {
-                    debug!("No available peers to ask for the model parameters at the moment");
+                    info!("No available peers to ask for the model parameters at the moment");
                     None
                 };
                 let _ = reply.send(peer);
             }
             PeerCommand::ReportSuccess { peer_id } => {
-                self.available_peers.push_back(peer_id);
-                debug!("Peer {peer_id} correctly provided the blob ticket");
+                if !self.available_peers.contains(&peer_id) {
+                    self.available_peers.push_back(peer_id);
+                } else {
+                    error!("Peer was already available but we tried to add it again");
```

Comment:
```suggestion
                    warn!("Peer was already available but we tried to add it again");
```

I guess it's a warn! since it's not fatal